mlrun/api/utils/clients/log_collector.py

```diff
@@ -203,12 +204,32 @@ async def has_logs(
         :param raise_on_error: Whether to raise an exception on error
         :return: Whether the log collector service has logs for the given run
         """
+
+        def _retryable_error(error_message: str) -> bool:
+            retryable_error_regex = [
+                # when multiple routines in the log collector service try to search the same directory,
+                # one of them can fail with this error
+                "readdirent.*resource temporarily unavailable"
+            ]
+            if any(re.match(regex, error_message) for regex in retryable_error_regex):
+                return True
```

Comment:
take a look at `mlrun.utils.helpers.verify_field_regex` we already have a similar implementation

Reply:
hmmm IMO this helper methods will just add unnecessary spammy logs if not matches , instead of just matching regex...